src/pages/[platform]/build-a-backend/server-side-rendering/nuxt/index.mdx

```diff
@@ -0,0 +1,571 @@
+const gqlServerClient = generateServerClient({ config: amplifyConfig });
+
+const getAmplifyAuthKeys = (lastAuthUser: string) =>
+  ['idToken', 'accessToken', 'refreshToken', 'clockDrift']
+    .map(
+      (key) =>
+        `CognitoIdentityServiceProvider.${userPoolClientId}.${lastAuthUser}.${key}`
+    )
+    .concat(lastAuthUserCookieName);
+
+// define the plugin
+export default defineNuxtPlugin({
+  name: 'AmplifyAPIs',
+  enforce: 'pre',
+  setup() {
+    // when setCookie is needed, set expirers as a year in this example
+    const expires = new Date();
+    expires.setDate(expires.getDate() + 365);
+
+    // get the last auth user cookie value
+    const lastAuthUserCookie = useCookie(lastAuthUserCookieName, {
+      sameSite: 'lax',
+      expires,
+      secure: true
+    });
+
+    // get all Amplify auth token cookie names
+    const authKeys = getAmplifyAuthKeys(lastAuthUserCookie.value!);
+
+    // create a key-value map of cookie name => cookie ref
+    const amplifyCookies = authKeys
+      .map((name) => ({
+        name,
+        cookieRef: useCookie(name, { sameSite: 'lax', expires, secure: true })
+      }))
+      .filter(
+        (item): item is { name: string; cookieRef: CookieRef } =>
+          item.cookieRef !== undefined
+      )
+      .reduce(
+        (result, current) => ({
+          ...result,
+          [current.name]: current.cookieRef
+        }),
+        {} as Record<string, CookieRef>
+      );
+
+    // create a key value storage based on the cookies
+    const keyValueStorage = createKeyValueStorageFromCookieStorageAdapter({
+      get(name) {
+        const cookieRef = amplifyCookies[name];
+
+        if (cookieRef && cookieRef.value) {
+          return { name, value: cookieRef.value };
+        }
+
+        return undefined;
+      },
+      getAll() {
+        return Object.entries(amplifyCookies).map(([name, cookieRef]) => {
+          return { name, value: cookieRef.value ?? undefined };
+        });
+      },
+      set(name, value) {
+        const cookieRef = amplifyCookies[name];
+        if (cookieRef) {
+          cookieRef.value = value;
+        }
+      },
+      delete(name) {
+        const cookieRef = amplifyCookies[name];
+
+        if (cookieRef) {
+          cookieRef.value = null;
+        }
+      }
+    });
+
+    // create a token provider
+    const tokenProvider = createUserPoolsTokenProvider(
+      amplifyConfig.Auth!,
+      keyValueStorage
+    );
+
+    // create a credentials provider
+    const credentialsProvider = createAWSCredentialsAndIdentityIdProvider(
+      amplifyConfig.Auth!,
+      keyValueStorage
+    );
+
+    // create the libraryOptions object
+    const libraryOptions: LibraryOptions = {
+      Auth: {
+        tokenProvider,
+        credentialsProvider
+      }
+    };
+
+    return {
+      provide: {
+        Amplify: {
+          Auth: {
+            fetchAuthSession: (options: FetchAuthSessionOptions) =>
+              runWithAmplifyServerContext(
```

Comment:
As above, the `runWithAmplifyServerContext` block is very important. Even if we've got runtime checks that will fail if we attempt to execute the provider without the server context, we should add some detail here about what the function does and why it's important.